ESP8266/ESP8266.cpp

```diff
@@ -562,6 +572,23 @@ void ESP8266::_oob_socket4_closed_handler()
     _socket_open[4] = 0;
 }
 
+void ESP8266::_connection_status_handler()
+{
+    char status[13];
+    if (_parser.recv("%12[^\"]\n", status)) {
+        if (strcmp(status, "CONNECTED\n") == 0)
+            _connection_status = NSAPI_STATUS_LOCAL_UP;
```

Comment:
Does this "local up" actually happen in ESP?
Does it even support link-local addresses?

If it does not, we probably wan't to skip this status and go from DISCONNECTED->GLOBAL UP